generators/generator-botbuilder/generators/app/templates/echo/bot.ts

```diff
@@ -1,41 +1,48 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
-const { ActivityTypes } = require('botbuilder');
+import { ActivityTypes, ConversationState, StatePropertyAccessor, TurnContext } from 'botbuilder';
+
 
 // Turn counter property
 const TURN_COUNTER_PROPERTY = 'turnCounterProperty';
 
 export class MyBot {
-    public countProperty;
-    public conversationState;
-  /**
-   *
-   * @param {ConversationState} conversation state object
-   */
-  constructor(conversationState) {
-    // Creates a new state accessor property.
-    // See https://aka.ms/about-bot-state-accessors to learn more about the bot state and state accessors.
-    this.countProperty = conversationState.createProperty(TURN_COUNTER_PROPERTY);
-    this.conversationState = conversationState;
-  }
-  /**
-   *
-   * @param {TurnContext} on turn context object.
-   */
-  async onTurn(turnContext) {
-    // See https://aka.ms/about-bot-activity-message to learn more about the message and other activity types.
-    if (turnContext.activity.type === ActivityTypes.Message) {
-      // read from state.
-      let count = await this.countProperty.get(turnContext);
-      count = count === undefined ? 1 : ++count;
-      await turnContext.sendActivity(`${count}: You said "${turnContext.activity.text}"`);
-      // increment and set turn counter.
-      await this.countProperty.set(turnContext, count);
-    } else {
-      await turnContext.sendActivity(`[${turnContext.activity.type} event detected]`);
+    
+    private readonly countAccessor: StatePropertyAccessor<number>;
+    private readonly conversationState: ConversationState;
+
+    /**
+     * 
+     * @param {ConversationState} conversation state object
+     */
+    constructor (conversationState: ConversationState) {
+        // Create a new state accessor property. 
+        // See https://aka.ms/about-bot-state-accessors to learn more about the bot state and state accessors.        
+        this.countAccessor = conversationState.createProperty(TURN_COUNTER_PROPERTY);
```

Comment:
This could be `this.countAccessor = conversationState.createProperty<number>(TURN_COUNTER_PROPERTY);`